zeebe/clients/java/src/test/java/io/camunda/zeebe/client/util/RestGatewayService.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.zeebe.client.util;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.junit5.WireMockRuntimeInfo;
+import io.camunda.zeebe.client.protocol.rest.BrokerInfo;
+import io.camunda.zeebe.client.protocol.rest.Partition;
+import io.camunda.zeebe.client.protocol.rest.Partition.HealthEnum;
+import io.camunda.zeebe.client.protocol.rest.Partition.RoleEnum;
+import io.camunda.zeebe.client.protocol.rest.TopologyResponse;
+import java.util.Arrays;
+import org.junit.jupiter.api.Assertions;
+
+public class RestGatewayService {
+
+  public static final ObjectMapper JSON_MAPPER = new ObjectMapper();
+
+  private final WireMockRuntimeInfo mockInfo;
+
+  public RestGatewayService(final WireMockRuntimeInfo mockInfo) {
+    this.mockInfo = mockInfo;
+  }
+
+  public void onTopologyRequest(
+      final int clusterSize,
+      final int partitionsCount,
+      final int replicationFactor,
+      final String gatewayVersion,
+      final BrokerInfo... brokers) {
+    try {
+      mockInfo
+          .getWireMock()
+          .register(
+              WireMock.get("/v1/topology")
```

Comment:
❓ Should we already start thinking on how we will handle the switch to the `v2` API?

Reply:
We'll move to `v2` with the job activation PR I'll open soon.
As far as the client here is concerned, we're still fine with `v1`.

Reply:
On second thought, we could also move to `v2` here now already, it would allow other teams to create proper tests now.